### ISSUE TITLE
Maximum entropy lemmatizer training files?

### ISSUE THREAD

**Issue by TheQuack45** [knuppe/SharpNL#31]
Hi, I don't know if this is a great place to ask this as it is more of a question with than an issue with SharpNL (which I am a huge fan of!), but I don't see anywhere else I could get help with this project. I'm developing a project in which I plan to make use of SharpNL's maximum entropy lemmatizer. I plan to use the lemmas to look in a thesaurus for synonyms of the original word and then conjugate one of those synonyms into the same form as the original word (why I am doing that is a long story). Stemming is not quite appropriate for this because it can't handle context well (I'm reading that 'meeting' could cause an issue for me, for example), so I would like to use lemmatization if possible.

However, as far as I can tell, while lemmatization seems to be built into SharpNL fully, there are no tests included with the testing project (which I've been using to learn how to set up the other tools in SharpNL) and no training files for lemmatization included. So I'm wondering if anyone has a training file available, or could explain to me how to write my own training file to use. I've been working through the lemmatizer training methods to try to figure out what it expects to write it myself, but I have no formal experience and very little informal experience with natural language processing or machine learning, so any help with this would be appreciated. Thanks!

**Comment by knuppe** [comment]
Hi, sorry for the late reply... 

Unfortunately I don't have any file to help you, but I think you may find good help in the mailing list of the original opennlp project [here](https://opennlp.apache.org/mail-lists.html). There are many people on these lists often using OpenNLP/NLP in general.

:v:

**Comment by TheQuack45** [comment]
Thanks for the help, I'll take a look there. I did manage to find a file to train it from, but when I serialize the model that results, and then try to deserialize it to use it in a program, it crashes with an InvalidOperationException, stating "End of Central Directory record could not be found." I'm still looking into it, seeing if I can figure it out myself, but if you have any ideas I'd appreciate the help; you'd certainly know more about this than I.

Opening up the .bin file containing the model is successful in 7-Zip, and shows a "lemmatizer.model" and a "manifest.properties" files, looking similar to how working model files look when opened in 7-Zip. The "manifest.properties" file also seems to be valid, but it does differ from other manifest.properties files that I looked at, missing some attributes and having other new ones compared to the others. So I'm not so sure what is wrong, but I'll keep looking into it.

If you have any ideas for my problem with this that would be great, but I don't want to take up your time, so don't worry about it if you don't have the time to.

Thanks!
